### DISCUSSION TITLE
How can you change the label of a button?

### DISCUSSION THREAD

**Discussion by nnmm** [Textualize/textual#1422] [Q&A] [answered]
This doesn't work:

```py
from textual.app import App, ComposeResult
from textual.widgets import Button

class MyApp(App):
    def compose(self) -> ComposeResult:
        yield Button('Change label', variant='success', id='btn')

    async def on_button_pressed(self, event: Button.Pressed) -> None:
        self.query_one('#btn').update('Whep')

if __name__ == '__main__':
    MyApp().run()
```

**Comment by nnmm** [comment, marked as answer]
Assign to the `label` property:

`self.query_one('#btn').label = 'Whep'`

**Reply by meshula** [reply]
When I do this, the UI to update. I can fetch the button again and read the label and see that it's updated, but the UI still shows the original label. Is there an additional thing that has to happen to cause the new label to display?

**Reply by meshula** [reply]
Answering my own question. assigning a string to the label alone doesn't do it. 

```py
                            label = "Patch " + url
                            patch_button.label = label
                            patch_button.update(label)
                            patch_button.refresh()
```

only setting the label, updating, and then refreshing seems to work 100% of the time.
